lib/ipam/ipam.go

```diff
@@ -195,9 +195,52 @@ func (c ipamClient) getBlockFromAffinity(ctx context.Context, aff *model.KVPair)
 	return b, nil
 }
 
-func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, attrs map[string]string, pools []net.IPNet, version ipVersion, host string) ([]net.IP, error) {
+func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, attrs map[string]string, requestedPools []net.IPNet, version ipVersion, host string) ([]net.IP, error) {
+	// Start by sanitizing the requestedPools.
+	pools := []net.IPNet{}
 
-	// Start by trying to assign from one of the host-affine blocks.  We
+	// Get all the configured pools.
+	enabledPools, err := c.pools.GetEnabledPools(version.Number)
+	if err != nil {
+		log.WithError(err).Errorf("Error reading configured pools")
```

Comment:
We create a context logger just a little bit below - I think you should move that above and then make sure all the logs are using `logCtx`